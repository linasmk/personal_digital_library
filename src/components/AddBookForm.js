/* ============ Dependencies ============= */
import React from "react";
import Modal from "react-modal";
import { AiOutlineClose} from "react-icons/ai";
/* ============== Redux =============== */

/* ============ Components ============== */
/* =============== Code ================= */
//Modal.setAppElement("#app");
class AddBookForm extends React.Component {
  state = {
    title: this.props.book ? this.props.book.title : "",
    author: this.props.book ? this.props.book.author : "",
    description: this.props.book ? this.props.book.description : "",
    publicationYear: this.props.book ? this.props.book.publicationYear : "",
    publisher: this.props.book ? this.props.book.publisher : "",
    publicationLength: this.props.book ? this.props.book.publicationLength : "",
    category: this.props.book ? this.props.book.category : "",
    url: this.props.book ? this.props.book.url : "",
    rating: this.props.book ? this.props.book.rating : "",
    imgLink: this.props.book ? this.props.book.imgLink : "",
    error: "",
  };
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({
      title,
    }));
  };
  onAuthorChange = (e) => {
    const author = e.target.value;
    this.setState(() => ({
      author,
    }));
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({
      description,
    }));
  };
  onPublicationYearChange = (e) => {
    const publicationYear = e.target.value;
    this.setState(() => ({
      publicationYear,
    }));
  };

  onPublicationLengthChange = (e) => {
    const publicationLength = e.target.value;
    this.setState(() => ({
      publicationLength,
    }));
  };

  onPublisherChange = (e) => {
    const publisher = e.target.value;
    this.setState(() => ({
      publisher,
    }));
  };

  onUrlChange = (e) => {
    const url = e.target.value;
    this.setState(() => ({
      url,
    }));
  };
  
  onCategoryChange = (e) => {
    const category = e.target.value;
    this.setState(() => ({
      category,
    }));
  };
  
  onRatingChange = (e) => {
    const rating = e.target.value;
    this.setState(() => ({
      rating,
    }));
  };
  onImgLinkChange = (e) => {
    const imgLink = e.target.value;
    this.setState(() => ({
      imgLink,
    }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title) {
      this.setState(() => ({
        error: "Title's name field should not stay empty!",
      }));
    } else {
      this.setState(() => ({
        error: "",
      }));
      this.props.onSubmit({
        title: this.state.title,
        author: this.state.author,
        description: this.state.description,
        publicationYear: this.state.publicationYear,
        publisher: this.state.publisher,
        publicationLength: this.state.publicationLength,
        category: this.state.category,
        url: this.state.url,
        rating: this.state.rating,
        imgLink: this.state.imgLink,
      });
      this.props.clearAddNewBookModal();
    }
  };
  render() {
    return (
      <Modal
        isOpen={this.props.addNewBookModal}
        onRequestClose={this.props.clearAddNewBookModal}
        contentLabel={"Add New Book"}
        closeTimeoutMS={800}
        className="modal"
        ariaHideApp={false}
      >
      <div className="top-bar h-top-bar-modal">
        <AiOutlineClose
            className="card__btn-close h-icon-styles"
            viewBox="100 50 800 880"
            onClick={this.props.clearAddNewBookModal}
          />
      </div>
        <h2 className="addbook__heading">Add New Book</h2>
        <form className="addbook__form" onSubmit={this.onSubmit}>
          {this.state.error && (
            <p className="addbook-form__warning">{this.state.error}</p>
          )}
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.title}
                onChange={this.onTitleChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Book Title</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.author}
                onChange={this.onAuthorChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Author</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationYear}
                onChange={this.onPublicationYearChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publication Year</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationLength}
                onChange={this.onPublicationLengthChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">
                  Publication Length
                </span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.publisher}
                onChange={this.onPublisherChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publisher</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                pattern="https://.*"
                autoComplete="off"
                value={this.state.url}
                onChange={this.onUrlChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Link to the Book</span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.category}
                onChange={this.onCategoryChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Category</span>
              </label>
            </div>

            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                min="1"
                max="10"
                required
                value={this.state.rating}
                onChange={this.onRatingChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Rating</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                autoComplete="on"
                value={this.state.imgLink}
                onChange={this.onImgLinkChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Image Link</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column-textarea">
              <textarea
                className="addbook__form_textarea"
                type="text"
                required
                maxLength="3000"
                value={this.state.description}
                onChange={this.onDescriptionChange}
              />
              <label htmlFor="title" className="addbook__form_label-textarea">
                <span className="addbook__form_content-textarea">
                  Book Description
                </span>
              </label>
            </div>
          </div>
          <div className="modal__button-bottom_wrapper">
            <button className="modal__button-bottom" type="submit">
              Add Book
            </button>
          </div>
        </form>
      </Modal>
    );
  }
}

export default AddBookForm;
