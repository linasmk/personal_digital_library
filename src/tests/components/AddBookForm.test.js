/* ===== App Dependencies ===== */
import React from "react";
import { shallow } from "enzyme";
import books from "../fixtures/books";
/* ========== Components ========== */
import AddBookForm from "../../components/AddBookForm";

/* ========== Code ========== */
test("Should render AddBookForm correctly", () => {
  const wrapper = shallow(<AddBookForm />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render AddBookForm with books data", () => {
  const wrapper = shallow(<AddBookForm book={books[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render error for invalid form submission", () => {
  const wrapper = shallow(<AddBookForm />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {},
  });
  expect(wrapper.state("error").length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test("Should set title on input change", () => {
  const value = "New title";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(0).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("title")).toBe(value);
});

test("Should set author on input change", () => {
  const value = "New author";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(1).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("author")).toBe(value);
});

test("Should set publicationYear on input change", () => {
  const value = "New publication year";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(2).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("publicationYear")).toBe(value);
});

test("Should set publicationLength on input change", () => {
  const value = "New publication length";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(3).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("publicationLength")).toBe(value);
});

test("Should set publisher on input change", () => {
  const value = "New publisher";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(4).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("publisher")).toBe(value);
});

test("Should set url on input change", () => {
  const value = "New url";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(5).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("url")).toBe(value);
});

test("Should set category on input change", () => {
  const value = "New category";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(6).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("category")).toBe(value);
});

test("Should set rating on input change", () => {
  const value = "New rating";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(7).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("rating")).toBe(value);
});

test("Should set imgLink on input change", () => {
  const value = "New imgLink";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("input").at(8).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("imgLink")).toBe(value);
});

test("Should set description on textarea change", () => {
  const value = "New description";
  const wrapper = shallow(<AddBookForm />);
  wrapper.find("textarea").at(0).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("description")).toBe(value);
});

test("Should call onSubmit prop for valid form submission", () => {
  const onSubmitSpy = jest.fn();
  const clearAddNewBookModalSpy = jest.fn();
  const wrapper = shallow(
    <AddBookForm
      book={books[0]}
      onSubmit={onSubmitSpy}
      clearAddNewBookModal={clearAddNewBookModalSpy}
    />
  );
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {},
  });

  expect(wrapper.state("error")).toBe("");
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    title: books[0].title,
    author: books[0].author,
  });
  expect(clearAddNewBookModalSpy).toBe("");
});

// test("Should call onSubmit prop for valid form submission", () => {
//   const onSubmitSpy = jest.fn();
//   const clearAddNewBookModalSpy = jest.fn();
//   onSubmitSpy();
//   clearAddNewBookModalSpy();
// });
