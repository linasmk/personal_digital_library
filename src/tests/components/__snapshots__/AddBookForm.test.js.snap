// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddBookForm correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={800}
  contentLabel="Add New Book"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="add__book_hading"
  >
    Add New Book
  </h2>
  <form
    className="addbook__form"
    onSubmit={[Function]}
  >
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Book Title
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Author
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Year
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Length
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publisher
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          pattern="https://.*"
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Link to the Book
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Category
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          max="10"
          min="1"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Rating
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="on"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Image Link
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column-textarea"
      >
        <textarea
          className="addbook__form_textarea"
          maxLength="3000"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label-textarea"
          htmlFor="title"
        >
          <span
            className="addbook__form_content-textarea"
          >
            Book Description
          </span>
        </label>
      </div>
    </div>
    <div
      className="modal__button-bottom_wrapper"
    >
      <button
        className="modal__button-bottom"
        type="submit"
      >
        Add Book
      </button>
    </div>
  </form>
  <button
    className="modal__button-top"
  >
    <span>
      Ã—
    </span>
  </button>
</Modal>
`;

exports[`Should render AddBookForm with books data 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={800}
  contentLabel="Add New Book"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="add__book_hading"
  >
    Add New Book
  </h2>
  <form
    className="addbook__form"
    onSubmit={[Function]}
  >
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value="Incognito: The Secret Lives of the Brain"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Book Title
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value="David Eagleman"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Author
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Year
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Length
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publisher
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          pattern="https://.*"
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Link to the Book
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Category
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          max="10"
          min="1"
          onChange={[Function]}
          required={true}
          type="number"
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Rating
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="on"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Image Link
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column-textarea"
      >
        <textarea
          className="addbook__form_textarea"
          maxLength="3000"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="addbook__form_label-textarea"
          htmlFor="title"
        >
          <span
            className="addbook__form_content-textarea"
          >
            Book Description
          </span>
        </label>
      </div>
    </div>
    <div
      className="modal__button-bottom_wrapper"
    >
      <button
        className="modal__button-bottom"
        type="submit"
      >
        Add Book
      </button>
    </div>
  </form>
  <button
    className="modal__button-top"
  >
    <span>
      Ã—
    </span>
  </button>
</Modal>
`;

exports[`Should render error for invalid form submission 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={800}
  contentLabel="Add New Book"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="add__book_hading"
  >
    Add New Book
  </h2>
  <form
    className="addbook__form"
    onSubmit={[Function]}
  >
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Book Title
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Author
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Year
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Length
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publisher
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          pattern="https://.*"
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Link to the Book
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Category
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          max="10"
          min="1"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Rating
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="on"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Image Link
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column-textarea"
      >
        <textarea
          className="addbook__form_textarea"
          maxLength="3000"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label-textarea"
          htmlFor="title"
        >
          <span
            className="addbook__form_content-textarea"
          >
            Book Description
          </span>
        </label>
      </div>
    </div>
    <div
      className="modal__button-bottom_wrapper"
    >
      <button
        className="modal__button-bottom"
        type="submit"
      >
        Add Book
      </button>
    </div>
  </form>
  <button
    className="modal__button-top"
  >
    <span>
      Ã—
    </span>
  </button>
</Modal>
`;

exports[`Should render error for invalid form submission 2`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={800}
  contentLabel="Add New Book"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="add__book_hading"
  >
    Add New Book
  </h2>
  <form
    className="addbook__form"
    onSubmit={[Function]}
  >
    <p
      className="addbook-form__warning"
    >
      Title's name field should not stay empty!
    </p>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Book Title
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Author
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Year
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publication Length
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Publisher
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="off"
          className="addbook__form_input"
          onChange={[Function]}
          pattern="https://.*"
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Link to the Book
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Category
          </span>
        </label>
      </div>
      <div
        className="addbook__form_column"
      >
        <input
          className="addbook__form_input"
          max="10"
          min="1"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="title"
        >
          <span
            className="addbook__form_content"
          >
            Rating
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column"
      >
        <input
          autoComplete="on"
          className="addbook__form_input"
          onChange={[Function]}
          required={true}
          title="URL should match the pattern: https://.*"
          type="url"
          value=""
        />
        <label
          className="addbook__form_label"
          htmlFor="url"
        >
          <span
            className="addbook__form_content"
          >
            Image Link
          </span>
        </label>
      </div>
    </div>
    <div
      className="addbook__form_row"
    >
      <div
        className="addbook__form_column-textarea"
      >
        <textarea
          className="addbook__form_textarea"
          maxLength="3000"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="addbook__form_label-textarea"
          htmlFor="title"
        >
          <span
            className="addbook__form_content-textarea"
          >
            Book Description
          </span>
        </label>
      </div>
    </div>
    <div
      className="modal__button-bottom_wrapper"
    >
      <button
        className="modal__button-bottom"
        type="submit"
      >
        Add Book
      </button>
    </div>
  </form>
  <button
    className="modal__button-top"
  >
    <span>
      Ã—
    </span>
  </button>
</Modal>
`;
